src/ai/defaultai.cc

```diff
@@ -5905,34 +5888,25 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 			}
 		} else if (bo.type == BuildingObserver::Type::kMine) {
 			bo.primary_priority = bo.max_needed_preciousness;
-			if (mines_per_type[bo.mines].total_count() == 0 &&
-			    site_needed_for_economy != BasicEconomyBuildingStatus::kDiscouraged) {
+			if ((mines_per_type[bo.mines].total_count() == 0 &&
+			     site_needed_for_economy != BasicEconomyBuildingStatus::kDiscouraged) ||
+			    (mines_per_type[bo.mines].finished == mines_per_type[bo.mines].total_count() &&
+			     bo.current_stats >
+			        static_cast<uint32_t>(85 + std::abs(management_data.get_military_number_at(129)) /
+			                                      10) &&
```

Comment:
To shorten the condition and improve readability, you should put this calculation in front of the `if`